### PR TITLE
Don't ignore alignment anymore in ArgType::make_indirect

### PR DESCRIPTION
**Pull request by nox**

### CODE REVIEW

**Comment by eddyb** [comment]
Let's see if this works!

@bors r+

**Comment by nox** [comment]
Mmmh, are there no test results at all? :(

**Comment by shepmaster** [comment]
Ping from triage, @nox ! You have review comments to address.

**Comment by pietroalbini** [comment]
Thank you for this PR @nox! Unfortunately we haven't heard from you on this in a while, so I'm closing the PR to keep things tidy. Don't worry though, if you'll have time again in the future please **reopen** this PR, we'll be happy to review it again!

**Comment by eddyb** [comment]
LGTM, but this turned into more or less a hack. r? @alexcrichton

**Comment by nox** [comment]
IMO even if we decide to still not emit the alignment attribute, we should at least keep the change to `librustc_target` and reset it to `None` in `librustc_codegen_llvm`.

**Comment by alexcrichton** [comment]
@eddyb hm I'm not really sure I know what's going on here at all, did you intend to r? me?

**Comment by eddyb** [comment]
@alexcrichton IIRC, I hit this bug in #45225, and you preferred disabling the `align` universally, instead of only on `i686-pc-windows` (which appeared to be the only platform which has the issue).
I pinged you in case you think the special-casing to `i686-pc-windows` is too hacky to add.

@nox Hey, do you think maybe the connection here is `byval`? Since many targets don't use it.

File: src/librustc_codegen_llvm/abi.rs [hunk 1]
```diff
@@ -476,6 +476,14 @@ impl<'a, 'tcx> FnTypeExt<'a, 'tcx> for FnType<'tcx, Ty<'tcx>> {
                 }
             }
 
+            // FIXME(nox): Pointee alignment for indirect byval arguments
+            // make LLVM very unhappy.
```
Comment (eddyb) [inline src/librustc_codegen_llvm/abi.rs]: Can you include the name of the test that fails?

**Comment by kennytm** [comment]
Two run-pass tests failed on `i686-msvc-2`.

<details>

```
failures:
---- [run-pass] run-pass\extern-pass-TwoU64s.rs stdout ----
error: test run failed!
status: exit code: 101
command: PATH="C:\projects\rust\build\i686-pc-windows-msvc\stage2\lib\rustlib\i686-pc-windows-msvc\lib;C:\projects\rust\build\i686-pc-windows-msvc\llvm\build\bin;C:\Program Files (x86)\Windows Kits\10\bin\x86;C:\Program Files (x86)\Windows Kits\10\bin\10.0.14393.0\x86;C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\amd64;C:\projects\rust\build\i686-pc-windows-msvc\stage0-tools\i686-pc-windows-msvc\release\deps;C:\projects\rust\build\i686-pc-windows-msvc\stage0-sysroot\lib\rustlib\i686-pc-windows-msvc\lib;C:\Program Files (x86)\Inno Setup 5;C:\Python27;C:\msys64\mingw32\bin;C:\msys64\usr\bin;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\Perl\site\bin;C:\Perl\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Program Files\7-Zip;C:\Program Files\Microsoft\Web Platform Installer;C:\Tools\GitVersion;C:\Tools\PsTools;C:\Program Files\Git LFS;C:\Program Files (x86)\Subversion\bin;C:\Program Files\Microsoft SQL Server\120\Tools\Binn;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\110\Tools\Binn;C:\Program Files (x86)\Microsoft SQL Server\120\Tools\Binn;C:\Program Files\Microsoft SQL Server\120\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\120\Tools\Binn\ManagementStudio;C:\Tools\WebDriver;C:\Program Files (x86)\Microsoft SDKs\TypeScript\1.4;C:\Program Files (x86)\Microsoft Visual Studio 12.0\Common7\IDE\PrivateAssemblies;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI\wbin;C:\Ruby193\bin;C:\Tools\NUnit\bin;C:\Tools\xUnit;C:\Tools\MSpec;C:\Tools\Coverity\bin;C:\Program Files (x86)\CMake\bin;C:\go\bin;C:\Program Files\Java\jdk1.8.0\bin;C:\Python27;C:\Program Files\nodejs;C:\Program Files (x86)\iojs;C:\Program Files\iojs;C:\Users\appveyor\AppData\Roaming\npm;C:\Program Files\Microsoft SQL Server\130\Tools\Binn;C:\Program Files (x86)\MSBuild\14.0\Bin;C:\Tools\NuGet;C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Program Files\Microsoft DNX\Dnvm;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn;C:\Program Files\Microsoft SQL Server\130\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\120\DTS\Binn;C:\Program Files (x86)\Apache\Maven\bin;C:\Python27\Scripts;C:\Tools\NUnit3;C:\Program Files\Mercurial;C:\Program Files\dotnet;C:\Tools\curl\bin;C:\Program Files\Amazon\AWSCLI;C:\Program Files (x86)\Microsoft SQL Server\140\DTS\Binn;C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\SQLDB\DAC\140;C:\Tools\vcpkg;C:\Program Files (x86)\Microsoft SQL Server\140\Tools\Binn;C:\Program Files\Microsoft SQL Server\140\Tools\Binn;C:\Program Files\Microsoft SQL Server\140\DTS\Binn;C:\Program Files\erl9.2\bin;C:\Program Files\Git\cmd;C:\Program Files\Git\usr\bin;C:\Program Files (x86)\Yarn\bin;C:\Program Files (x86)\NSIS;C:\Tools\Octopus;C:\Program Files\PowerShell\6.0.2;C:\ProgramData\chocolatey\bin;C:\Program Files (x86)\nodejs;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\LLVM\bin;C:\Users\appveyor\AppData\Local\Yarn\bin;C:\Users\appveyor\AppData\Roaming\npm;C:\Program Files\AppVeyor\BuildAgent;C:\projects\rust;C:\projects\rust\handle" "C:\\projects\\rust\\build\\i686-pc-windows-msvc\\test\\run-pass\\extern-pass-TwoU64s\\a.exe"
stdout:
------------------------------------------
------------------------------------------
stderr:
------------------------------------------
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `TwoU64s { one: 22, two: 23 }`,
 right: `TwoU64s { one: 94505926656, two: 98784247808 }`', C:\projects\rust\src/test\run-pass\extern-pass-TwoU64s.rs:30:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.
------------------------------------------
thread '[run-pass] run-pass\extern-pass-TwoU64s.rs' panicked at 'explicit panic', tools\compiletest\src\runtest.rs:3042:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.
---- [run-pass] run-pass\struct-return.rs stdout ----
error: test run failed!
status: exit code: 101
command: PATH="C:\projects\rust\build\i686-pc-windows-msvc\stage2\lib\rustlib\i686-pc-windows-msvc\lib;C:\projects\rust\build\i686-pc-windows-msvc\llvm\build\bin;C:\Program Files (x86)\Windows Kits\10\bin\x86;C:\Program Files (x86)\Windows Kits\10\bin\10.0.14393.0\x86;C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\amd64;C:\projects\rust\build\i686-pc-windows-msvc\stage0-tools\i686-pc-windows-msvc\release\deps;C:\projects\rust\build\i686-pc-windows-msvc\stage0-sysroot\lib\rustlib\i686-pc-windows-msvc\lib;C:\Program Files (x86)\Inno Setup 5;C:\Python27;C:\msys64\mingw32\bin;C:\msys64\usr\bin;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\Perl\site\bin;C:\Perl\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Program Files\7-Zip;C:\Program Files\Microsoft\Web Platform Installer;C:\Tools\GitVersion;C:\Tools\PsTools;C:\Program Files\Git LFS;C:\Program Files (x86)\Subversion\bin;C:\Program Files\Microsoft SQL Server\120\Tools\Binn;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\110\Tools\Binn;C:\Program Files (x86)\Microsoft SQL Server\120\Tools\Binn;C:\Program Files\Microsoft SQL Server\120\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\120\Tools\Binn\ManagementStudio;C:\Tools\WebDriver;C:\Program Files (x86)\Microsoft SDKs\TypeScript\1.4;C:\Program Files (x86)\Microsoft Visual Studio 12.0\Common7\IDE\PrivateAssemblies;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI\wbin;C:\Ruby193\bin;C:\Tools\NUnit\bin;C:\Tools\xUnit;C:\Tools\MSpec;C:\Tools\Coverity\bin;C:\Program Files (x86)\CMake\bin;C:\go\bin;C:\Program Files\Java\jdk1.8.0\bin;C:\Python27;C:\Program Files\nodejs;C:\Program Files (x86)\iojs;C:\Program Files\iojs;C:\Users\appveyor\AppData\Roaming\npm;C:\Program Files\Microsoft SQL Server\130\Tools\Binn;C:\Program Files (x86)\MSBuild\14.0\Bin;C:\Tools\NuGet;C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Program Files\Microsoft DNX\Dnvm;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn;C:\Program Files\Microsoft SQL Server\130\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\120\DTS\Binn;C:\Program Files (x86)\Apache\Maven\bin;C:\Python27\Scripts;C:\Tools\NUnit3;C:\Program Files\Mercurial;C:\Program Files\dotnet;C:\Tools\curl\bin;C:\Program Files\Amazon\AWSCLI;C:\Program Files (x86)\Microsoft SQL Server\140\DTS\Binn;C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\SQLDB\DAC\140;C:\Tools\vcpkg;C:\Program Files (x86)\Microsoft SQL Server\140\Tools\Binn;C:\Program Files\Microsoft SQL Server\140\Tools\Binn;C:\Program Files\Microsoft SQL Server\140\DTS\Binn;C:\Program Files\erl9.2\bin;C:\Program Files\Git\cmd;C:\Program Files\Git\usr\bin;C:\Program Files (x86)\Yarn\bin;C:\Program Files (x86)\NSIS;C:\Tools\Octopus;C:\Program Files\PowerShell\6.0.2;C:\ProgramData\chocolatey\bin;C:\Program Files (x86)\nodejs;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\LLVM\bin;C:\Users\appveyor\AppData\Local\Yarn\bin;C:\Users\appveyor\AppData\Roaming\npm;C:\Program Files\AppVeyor\BuildAgent;C:\projects\rust;C:\projects\rust\handle" "C:\\projects\\rust\\build\\i686-pc-windows-msvc\\test\\run-pass\\struct-return\\a.exe"
stdout:
------------------------------------------
a: bbbbbbbc
b: cccccccb
c: 1d
d: aaaaaaa9
------------------------------------------
stderr:
------------------------------------------
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `14757395258681310140`,
 right: `14757395258967641293`', C:\projects\rust\src/test\run-pass\struct-return.rs:42:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.
------------------------------------------
thread '[run-pass] run-pass\struct-return.rs' panicked at 'explicit panic', tools\compiletest\src\runtest.rs:3042:9
failures:
    [run-pass] run-pass\extern-pass-TwoU64s.rs
    [run-pass] run-pass\struct-return.rs
test result: FAILED. 2980 passed; 2 failed; 24 ignored; 0 measured; 0 filtered out
thread 'main' panicked at 'Some tests failed', tools\compiletest\src\main.rs:498:22
```

</details>

**Comment by alexcrichton** [comment]
Er sorry, but yes I still believe that avoiding optimizations on a tier 1 platform is too hacky and we should not land it. 90% of the time this means that there's a legitimate bug in LLVM or something like that and it just happens to show up on one platform. It's quite rare that this literally would only apply to i686 windows

**Comment by nox** [comment]
I changed it to ignore alignment data for indirect sret parameters on all platforms.

File: src/librustc_codegen_llvm/abi.rs [hunk 2]
```diff
@@ -476,6 +476,15 @@ impl<'a, 'tcx> FnTypeExt<'a, 'tcx> for FnType<'tcx, Ty<'tcx>> {
                 }
             }
 
+            // HACK(nox): Pointee alignment for indirect sret arguments make
+            // LLVM unhappy at least on i686-pc-windows, and that leads to some
+            // tests failing (namely extern-pass-TwoU64s and struct-return).
```
Comment (eddyb) [inline src/librustc_codegen_llvm/abi.rs]: I don't thinks this is well-placed, since it runs *before* the `adjust_for_cabi` call in `adjust_for_abi`.

Comment (nox) [reply src/librustc_codegen_llvm/abi.rs]: Where should it go, then?

**Comment by pietroalbini** [comment]
Ping from triage @nox! The build is failing and there are review comments to address.

**Pull request closed without merging**